src/python/pants/option/options_bootstrapper_test.py

```diff
@@ -3,103 +3,123 @@
 
 import os
 import unittest
+from functools import partial
 from textwrap import dedent
+from typing import Dict, List, Optional
 
 from pants.base.build_environment import get_buildroot
+from pants.option.option_value_container import OptionValueContainer
 from pants.option.options_bootstrapper import OptionsBootstrapper
 from pants.option.scope import ScopeInfo
 from pants.util.contextutil import temporary_dir, temporary_file, temporary_file_path
 
 
-class BootstrapOptionsTest(unittest.TestCase):
+class OptionsBootstrapperTest(unittest.TestCase):
 
-  def _do_test(self, expected_vals, config, env, args):
-    self._test_bootstrap_options(config, env, args,
-                                 pants_workdir=expected_vals[0],
-                                 pants_supportdir=expected_vals[1],
-                                 pants_distdir=expected_vals[2])
-
-  def _config_path(self, path):
+  def _config_path(self, path: Optional[str]) -> List[str]:
     if path is None:
       return ["--pants-config-files=[]"]
-    else:
-      return [f"--pants-config-files=['{path}']"]
-
-  def _test_bootstrap_options(self, config, env, args, **expected_entries):
+    return [f"--pants-config-files=['{path}']"]
+
+  def assert_bootstrap_options(
+    self,
+    *,
+    config: Optional[Dict[str, str]] = None,
+    env: Optional[Dict[str, str]] = None,
+    args: Optional[List[str]] = None,
+    **expected_entries,
+  ) -> None:
     with temporary_file(binary_mode=False) as fp:
       fp.write('[DEFAULT]\n')
       if config:
         for k, v in config.items():
           fp.write(f'{k}: {v}\n')
       fp.close()
 
-      args = args + self._config_path(fp.name)
-      bootstrapper = OptionsBootstrapper.create(env=env, args=args)
+      args = [*self._config_path(fp.name), *(args or [])]
+      bootstrapper = OptionsBootstrapper.create(env=env or {}, args=args)
       vals = bootstrapper.get_bootstrap_options().for_global_scope()
 
-      vals_dict = {k: getattr(vals, k) for k in expected_entries}
-      self.assertEqual(expected_entries, vals_dict)
-
-  def test_bootstrap_option_values(self):
-    # Check all defaults.
-    buildroot = get_buildroot()
-
-    def br(path):
-      # Returns the full path of the given path under the buildroot.
-      return os.path.join(buildroot, path)
+    vals_dict = {k: getattr(vals, k) for k in expected_entries}
+    self.assertEqual(expected_entries, vals_dict)
+
+  def test_bootstrap_seed_values(self) -> None:
+    def assert_seed_values(
+      *,
+      config: Optional[Dict[str, str]] = None,
+      env: Optional[Dict[str, str]] = None,
+      args: Optional[List[str]] = None,
+      workdir: Optional[str] = None,
+      supportdir: Optional[str] = None,
+      distdir: Optional[str] = None,
+    ) -> None:
```

Comment:
This inlined helper function replaces original calls to `_do_test()`. The main change is that we no longer pass a 3-tuple of the expected values. Instead, we use explicit keyword args `workdir`, `supportdir`, and `distdir`. I found this to make the tests easier to read and to have less code thanks to the default args.